Fix 'upload file' action to use new API endpoints

dallanwagz opened the pull request:
### Features
<!-- Delete this section if your PR does not add any features -->
List any features you're adding to this app (e.g. new actions, parameters, auth methods, etc.)
* Added an option to enable debug level logging
* Added the ability to send "ask question" and "ask question channel" actions in a thread
-

### Bug Fixes
<!-- Delete this section if your PR does not fix any bugs -->
Describe any bugs you've fixed in this app, and how they were fixed
* Fixed linting and compilation errors for flake8 compatibility
* Updated "upload file" action to use new backend API endpoints. The endpoints currently being used were deprecated last month. See here: https://docs.slack.dev/changelog/2024-04-a-better-way-to-upload-files-is-here-to-stay/
* Fixed the "ask question channel" action. It was missing the logic to periodicially check for a response to the question 
-

### Manual Documentation

<details><summary>
Have you made any changes that should be documented in manual_readme_content.md?
</summary><br />

The following changes require documentation in `manual_readme_content.md`:

- New, updated, or removed [REST handlers](https://docs.splunk.com/Documentation/SOAR/current/DevelopApps/RESTHandlers)
- New, updated, or removed authentication methods, especially complex methods like OAuth
- Compatibility considerations with respect to deployment types (e.g. actions that cannot be run on cloud or an automation broker)
</details>

- [x] I have verified that manual documentation has been updated where appropriate

### Other information
<!-- Delete this entire section if unused -->
<!-- Please add any other pertinent information you would like reviewers to know about your change -->

---
Please refer to our [Contribution Guide](https://github.com/splunk-soar-connectors/.github/blob/main/.github/CONTRIBUTING.md) for any questions on submitting a pull request.

Thanks for contributing!

athreya-splunk commented on release_notes/2.9.2.md:
```diff
@@ -0,0 +1,5 @@
+* Fixed linting and compilation errors for flake8 compatibility
+* Updated "upload file" action to use new backend API endpoints. The endpoints currently being used were deprecated last month. See here: https://docs.slack.dev/changelog/2024-04-a-better-way-to-upload-files-is-here-to-stay/
```
This feature was released couple of months back. Please sync with main and remove the "upload  file" action related changes.

dallanwagz replied:
Got it, removed.

athreya-splunk commented on slack_connector.py:
```diff
@@ -244,6 +246,26 @@ def __init__(self):
         self._permit_list = None
         self._permitted_users = None
 
+    def _debug_log(self, message, force=False):
```
Do we need this method? I believe this was added to test / get the logs for the newly added actions?

dallanwagz replied:
Good call out, it's not needed in the production app. I've removed it.

athreya-splunk commented:
Hey @dallanwagz - could you pls sync this PR with main and address couple of comments that I've added as well?

Commit:
fix: fix ask_question_channel polling and add thread support for ask actions

- Fixed _ask_question_channel to actually poll for responses; previously it
  returned immediately after sending the question without waiting for an answer
- Extracted shared polling logic into _poll_for_question_response helper
- Added parent_message_ts parameter to ask_question and ask_question_channel
  to support replying within a thread
- Added debug_logging config option for verbose logging
- Updated slack-sdk dependency to 3.39.0
- Fixed typo in debug message ("uestion" -> "Question")

Commit:
chore: update version to 2.10.2

Commit:
docs: add Splunk SOAR contribution process guide

Summarizes fork/branch/PR workflow, pre-commit requirements, versioning
conventions, PR template, and scope rules from the official contributing docs.

Commit:
chore: exclude internal tooling files from repo

Commit:
chore: ignore local audit tooling files

Commit:
refactor: remove debug_logging per maintainer feedback

Commit:
revert: restore slack_bot.py to upstream

Commit:
revert: remove .gitignore added locally

dallanwagz commented:
@athreya-splunk I've addressed your comments and adjusted my PR as requested. Please let me know what you think

dallanwagz committed:
Merge branch 'main' into dw_upload_file_fixes

athreya-splunk commented:
The app is being converted to use SOAR SDK. https://github.com/splunk-soar-connectors/slack/pull/69
We'll move the change to the sdkfied app once the PR merged.
